### PR TITLE
Support complex numbers in DG routines

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This completes the support for complex numbers in the DG code.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@wthrowe would you be able to take this one?

**Comment by nilsvu** [comment]
I also have a full integration test using the elliptic DG operator on a complex equation. I was going to add that in the next PR, but could push it here too if you prefer.

File: tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_NormalDotFlux.cpp [hunk 1]
```diff
@@ -37,30 +37,31 @@ void check_normal_dot_flux(const tnsr::i<DataVector, Dim>& normal,
   }
 }
 
-template <size_t Dim, typename Fr, typename Symm, typename... RemainingIndices>
+template <size_t Dim, typename Fr, typename DataType, typename Symm,
+          typename... RemainingIndices>
 void test_with_random_values(
     const DataVector& used_for_size,
-    Tensor<DataVector, Symm,
+    Tensor<DataType, Symm,
```
Comment (wthrowe) [inline tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_NormalDotFlux.cpp:45]: Call the test function with the new type.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_NormalDotFlux.cpp (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_NormalDotFlux.cpp:45]: done

**Comment by wthrowe** [comment]
I think you messed up rebasing or something.  You added a bunch of new commits that don't seem to compile.

**Comment by nilsvu** [comment]
Oh sorry here's the fixup

**Comment by wthrowe** [comment]
Looks good.  Squash.

**Pull request merged**